wallet-new/src/Cardano/Wallet/API/V1/Types.hs

```diff
@@ -1202,6 +1204,41 @@ instance BuildableSafeGen AddressValidity where
     buildSafeGen _ AddressValidity{..} =
         bprint ("{ valid="%build%" }") isValid
 
+-- | An address is either recognised as "ours" or not. An address that is not
+--   recognised may still be ours e.g. an address generated by another wallet instance
+--   will not be considered "ours" until the relevant transaction is confirmed.
+--
+--   In other words, `AddressNotRecognised` makes an inconclusive statement about
+--   an address, whereas `AddressIsOurs` is unambiguous.
+data AddressIsOurs = AddressIsOurs
+                   | AddressNotRecognised
+    deriving (Show, Eq, Generic, Ord)
+
+addressIsOursFromUsed :: Bool -> V1 AddressIsOurs
+addressIsOursFromUsed used =
+    V1 $ if used then AddressIsOurs else AddressNotRecognised
+
+instance ToJSON (V1 AddressIsOurs) where
+    toJSON (V1 AddressIsOurs)        = String "is-ours"
+    toJSON (V1 AddressNotRecognised) = String "not-recognised"
```

Comment:
We have an (alas unwritten) conversion that we use snake case for enums, so it would be "is_ours", for example.

Reply:
That can be done generically though with some helpers written by hand or taken from https://hackage.haskell.org/package/aeson-casing-0.1.0.5/docs/Data-Aeson-Casing.html 

We probably already have a few stuff doing this.